calculator.cc

```diff
@@ -6,17 +6,33 @@
 
 namespace dev {
 
+bool Calculator::isValidExpression(std::vector<std::string> tokens) {
+    int noOfOperands = 0;
+    for (auto token : tokens) {
+        if ((noOfOperands < 2) && (!Operand::isOperand(token))) {
+            throw InvalidOperandException();
+        }
+
+        if (Operand::isOperand(token)) {
+            noOfOperands++;
+        } else {
+            noOfOperands--;
+        }
+    }
+    return (noOfOperands == 1);
```

Comment:
## 1.  Contract programming
 - Use **assert** for contract programming ( to check that there is no bus in the way how the code is called)

## 2. Invalid user input 
 - Use exceptions and `try` for validating user input. 

## 3. Success flow of the program 
 - Use if/else conditions